squash_bot/list_timetable/commands.py

```diff
@@ -0,0 +1,179 @@
+import typing
+from urllib.parse import urljoin
+from datetime import datetime
+from dataclasses import dataclass
+
+import enum
+import requests
+
+from squash_bot.core import command as _command
+from squash_bot.core import command_registry, lambda_function
+from squash_bot.core.data import dataclasses as core_dataclasses
+from squash_bot.settings import base as settings_base
+
+
+TIMETABLE_API_DATETIME_FORMAT = "YYYY-MM-DDTHH:mm:ss+00:00"
+TIMETABLE_HEADER_DATETIME_FORMAT = "%d-%m"
+TIMETABLE_RESULT_DATETIME_FORMAT = TIMETABLE_HEADER_DATETIME_FORMAT + " HH:mm"
+
+RESULTS = "Results"
+AVAILABLE_SLOTS = "AvailableSlots"
+START = "start"
+RESOURCE_SCHEDULE_ID = "ResourceScheduleId"
+
+
+class ListTimetableOptionType(enum.Enum):
+    FROM_DATE = "from-date"
+    DAYS = "days"
+    TIME_OF_DAY = "time-of-day"
+
+
+class TimeOfDayType(enum.Enum):
+    MORNING = "Morning"
+    AFTERNOON = "Afternoon"
+    EVENING = "Evening"
+    ANY = "Any"
+
+
+@dataclass
+class TimetableResult:
+    start_datetime: datetime
+    available_slots: int
+    schedule_id: int
+
+    def __str__(self):
+        return f"{datetime.strftime(DISCORD_RESPONSE_DATETIME_FORMAT)}: {available_slots} slots available"
+
+
+@command_registry.registry.register
+class ListTimetableCommand(_command.Command):
+    name = "list-timetable"
+    description = "List the timetable between two datetimes"
+    options = (
+        _command.CommandOption(
+            name=ListTimetableOptionType.FROM_DATE.value,
+            description="Datetime string in the format DD-MM-YY",
+            type=_command.CommandOptionType.STRING,
+            required=True,
+        ),
+        _command.CommandOption(
+            name=ListTimetableOptionType.DAYS.value,
+            description="The amount of days to check the timetable for",
+            type=_command.CommandOptionType.INTEGER,
+            required=True,
+        ),
+        _command.CommandOption(
+            name=ListTimetableOptionType.TIME_OF_DAY.value,
+            description="The time of day to check the timetable for. Default = Any",
+            type=_command.CommandOptionType.STRING,
+            required=False,
+        )
+    )
+
+    def _handle(
+        self,
+        options: dict[str, typing.Any],
+        base_context: dict[str, typing.Any],
+        guild: core_dataclasses.Guild,
+        user: core_dataclasses.User,
+    ) -> dict[str, typing.Any]:
+        # Check from-date is valid
+        try:
+            from_date_str = options[ListTimetableOptionType.FROM_DATE.value]
+            from_date = datetime.strptime(options[ListTimetableOptionType.FROM_DATE.value], DISCORD_COMMAND_DATETIME_FORMAT)
+            from_date = from_date.replace(year=datetime.utcnow().year)
```

Comment:
Year being 'replaced' because the default will likely be 1970.
That would be the only time that you would do that - and would we ever want to play squash on Jan 1st?

Reply:
>and would we ever want to play squash on Jan 1st?

Why write something we know is broken when doing something that works is pretty easy
https://github.com/AledWatkins/squash-bot/blob/7a482bd69325103e9ba43ed490dbc6b2a5466e2c/squash_bot/match_tracker/commands.py#L149

Reply:
This has been removed now anyway since we're using datetime.now()